airbyte-cdk/python/airbyte_cdk/sources/declarative/types.py

```diff
@@ -51,3 +51,67 @@ def __eq__(self, other: object) -> bool:
 
     def __ne__(self, other: object) -> bool:
         return not self.__eq__(other)
+
+
+class PerPartitionStreamSlice(StreamSlice):
```

Comment:
I think reading through this refactor, I feel like we're now passing `PerPartitionStreamSlice` across pretty much everywhere. And I agree with the decision. A generic map is what has led to this specific bug. Is it crazy to say we should actually just call this entire object `StreamSlice` since the name makes it sound more specialized than it actually is?

Reply:
I like it. I also feel the same. `PerPartitionStreamSlice` is actually a `StreamSlice` where a partition or cursor information can be defined or not which applies to our domain for any stream slice that is created